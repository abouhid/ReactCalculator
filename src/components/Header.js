import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <nav>
    <Link to="/">
      <h1>Home</h1>
    </Link>
    <Link to="/calculator">
      <h1>React Calculator</h1>
    </Link>
    <Link to="/quotes">
      <h1>Quotes</h1>
    </Link>
  </nav>
);

export default Header;
