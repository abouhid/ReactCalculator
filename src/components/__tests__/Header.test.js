/* eslint-disable react/forbid-foreign-prop-types */
import React from 'react';
import Header from '../Header';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('Header', () => {
  it('Tests the rendering of components', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
