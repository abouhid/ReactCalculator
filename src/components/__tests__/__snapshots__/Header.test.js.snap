// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests the rendering of components 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Home
    </h1>
  </a>
  <a
    href="/calculator"
    onClick={[Function]}
  >
    <h1>
      React Calculator
    </h1>
  </a>
  <a
    href="/quotes"
    onClick={[Function]}
  >
    <h1>
      Quotes
    </h1>
  </a>
</nav>
`;

exports[`Test suits for MyComponentWithLink should match with snapshot 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Home
    </h1>
  </a>
  <a
    href="/calculator"
    onClick={[Function]}
  >
    <h1>
      React Calculator
    </h1>
  </a>
  <a
    href="/quotes"
    onClick={[Function]}
  >
    <h1>
      Quotes
    </h1>
  </a>
</nav>
`;
